src/webview/wkwebview/synthetic_mouse_events.rs

```diff
@@ -0,0 +1,123 @@
+use super::NSString;
+use cocoa::appkit::{NSEvent, NSEventModifierFlags, NSEventType, NSView};
+use cocoa::base::{id, nil};
+use objc::{
+  declare::ClassDecl,
+  runtime::{Object, Sel},
+};
+use std::{ffi::c_void, ptr::null};
+
+pub unsafe fn setup(decl: &mut ClassDecl) {
+  decl.add_method(
+    sel!(otherMouseDown:),
+    other_mouse_down as extern "C" fn(&Object, Sel, id),
+  );
+  decl.add_method(
+    sel!(otherMouseUp:),
+    other_mouse_up as extern "C" fn(&Object, Sel, id),
+  );
+}
+
+extern "C" fn other_mouse_down(this: &Object, _sel: Sel, event: id) {
+  unsafe {
+    if event.eventType() == NSEventType::NSOtherMouseDown {
+      let webview = this.get_ivar::<id>("webview");
+      if *webview != nil {
+        let button_number = event.buttonNumber();
+        match button_number {
+          4 => {
+            let js = create_js_mouse_event(*webview, event, true, true);
+            let _: id = msg_send![*webview, evaluateJavaScript:NSString::new(&js) completionHandler:null::<*const c_void>()];
+          }
+          5 => {
+            let js = create_js_mouse_event(*webview, event, true, false);
+            let _: id = msg_send![*webview, evaluateJavaScript:NSString::new(&js) completionHandler:null::<*const c_void>()];
+          }
+
```

Comment:
I have no idea if this is correct, I couldn't what the values of `even.buttonNumber` means so I speculated they would be 4 & 5 but please do test this and maybe through a dbg! here to see what the values are actually. 

Also I am not sure if `x` and `y` positions are correct so please test that too and try to compare it with the same event in chrome or something.

cc @pewsheen @goenning

Reply:
The button numbers are 3 & 4.

`x` is correct, but `y` is upside down.

Reply:
thanks, please give it another test and feel free to modify the PR directly with any fixes you do

Reply:
Tested, and it looks good!

Reply:
thanks @pewsheen